Language/Expressions/Constants/equals_expression_t03.dart

```diff
@@ -9,22 +9,21 @@
 /// • An expression of one of the forms e1 == e2 or e1 != e2 where e1 and e2 are
 ///   constant expressions that evaluate to a numeric, string or boolean value
 ///   or to null.
-/// @description Checks that it is a compile-time error when an expression of
-/// the form e1 == e2 where e1 does not evaluate to a numeric, string or boolean
-/// value, is used to initialize a constant variable.
+/// @description Checks that it is not an time error when an expression of
+/// the form e1 == e2 or e1 != e2 where e1 does not evaluate to a numeric,
+/// string or boolean value, is used to initialize a constant variable.

```

Comment:
I think we need to express this in a slightly more direct way (the negations are piling up). The actual rule today is that `e1 == e2` is constant if `e1` and `e2` are constant, `e1` evaluates to an object of type `double` or an object with primitive equality, or `e2` is null (so `e1 == null` is constant for _any_ constant `e1`).
```suggestion
/// @description Checks that it is not a compile-time error when an expression of
/// the form `e1 == e2` or `e1 != e2` is used to initialize a constant variable, and
/// `e1` does not evaluate to a numeric, string or boolean value, as long as the
/// receiver `e1` has primitive equality or is a `double`.
```

I moved the long constraint to the end; English sentences tend to avoid having anything large (like 'where ... value') between the noun phrase and the verb phrase.